File: batches_javascript/batch_10_code_reordering/ref.js
function computeStats(data) {
    let mx = data[0], mn = data[0], total = 0;
    for (const v of data) {
        if (v > mx) mx = v;
        if (v < mn) mn = v;
        total += v;
    }
    const avg = data.length > 0 ? total / data.length : 0;
    return { avg, max: mx, min: mn };
}

const d = [10, 20, 30, 40, 50];
console.log(computeStats(d));

File: batches/batches/batch_10_code_reordering/ref.py
"""
Reference - Original code for Batch 10: Code Reordering
Computes stats for a list.
"""


def compute_stats(data):
    total = sum(data)
    n = len(data)
    avg = total / n if n else 0
    mx = max(data)
    mn = min(data)
    return avg, mx, mn


if __name__ == "__main__":
    d = [10, 20, 30, 40, 50]
    print(compute_stats(d))
